### PR TITLE
Fix some hlint warnings - part 2

### PR DESCRIPTION
**Pull request by jhrcek**
Second batch of hopefully non-controversial hlint fixes (as discussed in https://github.com/haskell/hackage-server/pull/1056#issuecomment-1097855098).

### CODE REVIEW

File: src/Distribution/Server/Features/BuildReports/BuildReports.hs [hunk 1]
```diff
@@ -89,8 +89,8 @@ data PkgBuildReports = PkgBuildReports {
     buildStatus :: !BuildStatus
 } deriving (Eq, Typeable, Show)
 
-data BuildReports = BuildReports {
-    reportsIndex :: !(Map.Map PackageId PkgBuildReports)
+newtype BuildReports = BuildReports {
```
Comment (jhrcek) [inline src/Distribution/Server/Features/BuildReports/BuildReports.hs:92]: Of all the "use newtype" suggestions I only applied those where the unrerlying field was strict, so as not to change strictness properties of any of the single-constructor single-field datatypes.

**Review by andreasabel — CHANGES_REQUESTED** [review]
I would abstain from the `data -> newtype` conversions. (see my comment)
Everything else has my 100% approval!

File: src/Distribution/Server/Features/BuildReports/BuildReports.hs (hunk 1 above)
Comment (jhrcek) [reply src/Distribution/Server/Features/BuildReports/BuildReports.hs]: Ok, makes sense. All the newtype changes reverted.

**Comment by jhrcek** [comment]
@andreasabel friendly ping :smile_cat:

**Review by gbaz — APPROVED** [review]

**Review by andreasabel — COMMENTED** [review]
Thanks for update!  Approved!

**Pull request merged**